### ISSUE TITLE
Feature: result comparison

### ISSUE THREAD

**Issue by AriPerkkio** [AriPerkkio/eslint-remote-tester#12]
This feature should provide developers confidence when making changes to existing rules. Goal is to minimize manual work. 

- New configuration option: `compare: true|false`. Default to false.

Results of previous scan should be temporarily saved when `compare: true` is set. Once the scan is complete the results will be compared with previous ones and diff should be highlighted. **Developers should easily see whether their code changes fixed the actual issue, triggered any new false positives or caused linter crashes.**

Currently this has been possible by manually moving the results and diffing them with new ones. 

Expected local environment development workflow:
 1. Collect initial results by running the scan
 2. Make code changes to specific rule
 3. Run scan again, now with `compare: true`
 4. See how linter acts with new changes 

Expected CI development workflow, on pull requests making changes to existing rules:
1. New PR is open and ready to merge
2. Collect initial results by running the scan on `master` branch
3. Checkout to work/feature branch
4. Run scan again, now with `compare: true`
5. Send result changes to PR thread

Possible issues:
- Prevent mixing various settings between compared scans, e.g. `resultParser`, `rulesUnderTesting`, `repositories`, `extensions`
- Handle `compare: true` when previous results are not available
